crates/bevy_app/src/app.rs

```diff
@@ -43,6 +43,15 @@ pub struct App {
 }
 
 impl Default for App {
+    /// Implements the [`Default`] trait for [`App`]
+    ///
+    /// It creates an [`empty`](./struct.App.html#method.empty) [`App`].  
+    /// It then [adds default stages](./struct.App.html#method.add_default_stages) and [adds]('./struct.App.html#method.add_event`) the [`AppExit`] event.  
+    /// It finally [adds](./struct.App.html#method.add_system_to_stage) the [`CoreStage::Last`] stage with the `bevy::ecs::world::World::clear_trackers()` exclusive system.
```

Comment:
I'm not sure I like this level of granularity in API docs: I worry that it will be irrelevant to most users, and prone to going stale.

If the source code wasn't available I'd be much more in favor of including this, but the 1% who cares can quickly check the source.